DigitalOffice/Shared/Kernel/CustomExceptionHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Kernel.Exceptions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Diagnostics;
+using System.Text.Json;
+
+namespace Kernel
+{
+    /// <summary>
+    /// Represents exception handler middleware. Provides method for handle exceptions.
+    /// </summary>
+    public static class CustomExceptionHandler
+    {
+        /// <summary>
+        /// Handle exceptions.
+        /// </summary>
+        /// <param name="context">Specified http context.</param>
+        /// <returns>Asynchronous operation for handling exceptions.</returns>
+        public static async Task HandleCustomException(HttpContext context)
+        {
+            var exception = context.Features.Get<IExceptionHandlerPathFeature>()?.Error;
+            
+            var errorResponse = new ErrorResponse
+            {
+                UtcTime = DateTime.UtcNow,
+            };
+
+            if (exception is BaseException baseException)
+            {
+                context.Response.StatusCode = baseException.StatusCode;
+                errorResponse.Header = baseException.Header;
+                errorResponse.Message = baseException.Message;
+            }
+            else
+            {
+                context.Response.StatusCode = (int) HttpStatusCode.InternalServerError;
```

Comment:
Why Internal in another block?

Reply:
Because in case of an error not inherited from the BaseError class, we cannot report anything more specific. As far as I know, Internal Server Error is used in such cases.

Reply:
Ok, good for me at this moment